conference/react-js/src/containers/SignInForm/index.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { PureComponent } from 'react';
+
+class SignInForm extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      isValidCredentials: true,
+    };
+
+    this.emailChangeHandler = this.emailChangeHandler.bind(this);
+    this.submitHandler = this.submitHandler.bind(this);
+    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
+  }
+
+  submitHandler(event) {
+    // simulate bad credentials
+    this.setState({
+      isValidCredentials: false,
+    });
+
+    event.preventDefault();
+  }
+
+  emailChangeHandler(event) {
+    const target = event.target;
+
+    this.setState({
+      email: target.value,
+      isValidCredentials: true,
+    });
+  }
+
+  passwordChangeHandler(event) {
+    const target = event.target;
+
+    this.setState({
+      password: target.value,
+      isValidCredentials: true,
+    });
+  }
+
+  render() {
+    return (
+      <div className="sign-in-wrapper">
+        <form className="sign-in" onSubmit={this.submitHandler}>
+          <h2 className="form-title sign-in__title">sign in</h2>
+          <label
+            htmlFor="sign-in-email"
+            className="sign-in__label form-label"
+          >
+            email:
+          </label>
+          <input
+            type="email"
+            name="email"
+            id="sign-in-email"
+            className={this.state.isValidCredentials ?
```

Comment:
[Classnames](https://www.npmjs.com/package/classnames) package helps you to handle conditional classes